File: front/src/api/payment.ts
import request from '@/utils/request'

import type { OrderPaymentPayload, OrderRefundPayload } from './order'

export interface PaymentDetailParams {
  orderNo: string
}

export interface PaymentStatusPayload {
  success: boolean
  paid: boolean
  payment: OrderPaymentPayload
  orderStatus: number
  paidAt: string
  checkoutSessionId: string
}

export interface PaymentStatusParams {
  orderNo?: string
  paymentNo?: string
  channel?: string
}

export interface PaymentRequestOptions {
  silentError?: boolean
}

export interface PaymentDetailPayload {
  payment: OrderPaymentPayload
  refund: OrderRefundPayload
  orderStatus: number
  paidAt: string
}

export function getPaymentStatus(data: PaymentStatusParams, options: PaymentRequestOptions = {}) {
  return request.post<PaymentStatusPayload, PaymentStatusPayload>(
    '/payment/status',
    data,
    {
      silentError: options.silentError,
    } as any,
  )
}

export function getPaymentDetail(params: PaymentDetailParams) {
  return request.get<PaymentDetailPayload, PaymentDetailPayload>('/payment/detail', {
    params,
  })
}

File: server/app/api/internal/types/payment.go
package types

type PaymentDetailReq struct {
	OrderNo string `form:"orderNo" json:"orderNo"`
}

type PaymentTradeCheckReq struct {
	OrderNo   string `json:"orderNo,optional"`
	PaymentNo string `json:"paymentNo,optional"`
	Channel   string `json:"channel,optional"`
}

type PaymentDetailResp struct {
	Payment     OrderPaymentInfo `json:"payment"`
	Refund      OrderRefundInfo  `json:"refund"`
	OrderStatus int32            `json:"orderStatus"`
	PaidAt      string           `json:"paidAt"`
}

type PaymentTradeCheckResp struct {
	Success           bool             `json:"success"`
	Paid              bool             `json:"paid"`
	Payment           OrderPaymentInfo `json:"payment"`
	OrderStatus       int32            `json:"orderStatus"`
	PaidAt            string           `json:"paidAt"`
	CheckoutSessionId string           `json:"checkoutSessionId"`
}
